jgrapht-core/src/main/java/org/jgrapht/alg/chordal/ChordalGraphInspector.java

```diff
@@ -0,0 +1,191 @@
+                    bucketList.updateLabel(unvisitedNeighbour, vertexNum);
+                }
+            }
+            return lexBfsOrder;
+        } else {
+            return new ArrayList<>();
+        }
+    }
+
+    /**
+     * Returns yet unvisited by the lexicographical breadth-first search neighbours of the {@code vertex}
+     *
+     * @param graph      the graph, on which LexBFS is performed
+     * @param bucketList data structure, that backs the algorithm up
+     * @param vertex     the vertex, whose neighbours are explored
+     * @return the set of yet unvisited neighbours of the {@code vertex}
+     */
+    private Set<V> getUnvisitedNeighbours(Graph<V, E> graph, BucketList<V> bucketList, V vertex) {
+        Set<V> unmapped = new HashSet<>();
+        Set<E> edges = graph.edgesOf(vertex);
+        for (E edge : edges) {
+            V source = graph.getEdgeSource(edge);
+            V target = graph.getEdgeTarget(edge);
+            V dest = source.equals(vertex) ? target : source;
+            if (bucketList.containsBucketWith(dest)) {
+                unmapped.add(dest);
+            }
+        }
+        return unmapped;
+    }
+
+    /**
+     * Checks whether the vertices in the {@code vertexOrder} are in perfect elimination order.
+     * Returns false, is {@code graph} isn't chordal.
+     *
+     * @param graph       the graph whose vertices are put in {@code vertexOrder}
+     * @param vertexOrder the sequence of vertices of {@code graph}
+     * @return true if the {@code graph} is chordal and the vertices in {@code vertexOrder} are in
+     * perfect elimination order
+     */
+    public boolean isPerfectEliminationOrder(Graph<V, E> graph, List<V> vertexOrder) {
+        Set<V> graphVertices = graph.vertexSet();
+        if (graphVertices.size() == vertexOrder.size() && graphVertices.containsAll(vertexOrder)) {
+            Map<V, Integer> map = new HashMap<>();
+            int i = 0;
+            for (V vertex : vertexOrder) {
+                map.put(vertex, i);
+                ++i;
+            }
+            return isPerfectEliminationOrder(graph, vertexOrder, map);
+        } else {
+            return false;
+        }
+    }
+
+    /**
+     * Checks whether the vertices in the {@code vertexOrder} are in perfect elimination order.
+     * Returns false, is {@code graph} isn't chordal.
+     *
+     * @param graph       the graph whose vertices are put in {@code vertexOrder}
+     * @param vertexOrder the sequence of vertices of {@code graph}
+     * @param map         maps every vertex in {@code graph} to its position in {@code vertexOrder}, is used for constant-time lookups
+     * @return true if the {@code graph} is chordal and the vertices in {@code vertexOrder} are in
+     * perfect elimination order
+     */
+    private boolean isPerfectEliminationOrder(Graph<V, E> graph, List<V> vertexOrder, Map<V, Integer> map) {
+        Set<Pair<V, V>> testEdges = new HashSet<>();
```

Comment:
The reason was not to check the presence of the same edge multiple times.
For example, if we have a 4-vertex graph G = (V, E): V = {1,2,3,4}, E = {{1,2},{1,3},{1,4},{2,3},{2,4}}; and vertexOrder = (1,2,3,4), then:
* first and second iterations give no edges;
* third and fourth iteration give us the same edge {1,2} two times.

Clearly, there are examples, where the same edge can be encountered more than 2 times. And it can be useful to have this buffer in cases, where we can't check the presence of an edge in constant time. But then I found out that FastLookupUndirectedSpecifics is used rather than UndirectedSpecifics (correct me if I'm wrong), which allows constant-time lookups.
So, now both edge addition to buffer and in-time presence check take constant time. Buffering edges increases space complexity to O(|E|). At the same time, it slows down the method in those cases, when we could almost immediately say, that vertices in vertexOrder aren't in the perfect elimination order by revealing a missing edge. 
So, yeah, there's no need to buffer those edges.